diff --git a/openbsc/configure.ac b/openbsc/configure.ac
index 24dbc30..a38462e 100644
--- a/openbsc/configure.ac
+++ b/openbsc/configure.ac
@@ -13,7 +13,7 @@ dnl checks for programs
 AC_PROG_MAKE_SET
 AC_PROG_CC
 AC_PROG_INSTALL
-AC_PROG_RANLIB
+LT_INIT
 
 dnl checks for libraries
 AC_SEARCH_LIBS([dlopen], [dl dld], [LIBRARY_DL="$LIBS";LIBS=""])
diff --git a/openbsc/src/osmo-nitb/Makefile.am b/openbsc/src/osmo-nitb/Makefile.am
index d3b97f8..3b7cc8d 100644
--- a/openbsc/src/osmo-nitb/Makefile.am
+++ b/openbsc/src/osmo-nitb/Makefile.am
@@ -11,7 +11,6 @@ osmo_nitb_SOURCES = bsc_hack.c
 osmo_nitb_LDADD = \
 		$(top_builddir)/src/libbsc/libbsc.a \
 		$(top_builddir)/src/libmsc/libmsc.a \
-		$(top_builddir)/src/libbsc/libbsc.a \
 		$(top_builddir)/src/libtrau/libtrau.a \
 		$(top_builddir)/src/libcommon/libcommon.a \
 		-ldbi $(LIBCRYPT)				   \
diff --git a/openbsc/tests/abis/Makefile.am b/openbsc/tests/abis/Makefile.am
index c2e38de..6dc29f8 100644
--- a/openbsc/tests/abis/Makefile.am
+++ b/openbsc/tests/abis/Makefile.am
@@ -11,7 +11,6 @@ abis_test_SOURCES = abis_test.c
 abis_test_LDADD = \
 		$(top_builddir)/src/libbsc/libbsc.a \
 		$(top_builddir)/src/libcommon/libcommon.a \
-		$(top_builddir)/src/libbsc/libbsc.a \
 		$(top_builddir)/src/libtrau/libtrau.a \
 		$(LIBOSMOCORE_LIBS) $(LIBOSMOABIS_LIBS) \
 		$(LIBOSMOGSM_LIBS)
diff --git a/openbsc/tests/db/db_test.c b/openbsc/tests/db/db_test.c
index a02d1f8..ad230e0 100644
--- a/openbsc/tests/db/db_test.c
+++ b/openbsc/tests/db/db_test.c
@@ -256,4 +256,4 @@ int main()
 }
 
 /* stubs */
-void vty_out() {}
+static void vty_out() {}
